B dplan 16178 phase iteration maxrange fix

kijewskidemos opened the pull request:
### Ticket
[DPLAN-16178]
(https://demoseurope.youtrack.cloud/issue/DPLAN-16178) Nicht möglich um Grundeinstellungen zu speichern

### Description: 

If an interation number (Durchgangsnummer) higher than a smallint (~65.000) is being added to the field, an error is being thrown. This PR fixes this issue by validating that the given number is not higher than a smallint, doing so with a regex validation in the frontend and an additional check in the backend

### How to review/test

Log in as Fachplanung-Admin, Institutions-Koordination, go to Verfahren --> Grundeinstellungen --> Verfahrensschritt Institutionen and Verfahrensschritt Öffentlichkeit: add a number higher than 65535 . A error message should now pop up. -->

### PR Checklist
<!-- Reminders for handling PRs 

Delete the checkbox if it doesn't apply/isn't necessary. -->

- [ ] Create/Update tests
- [ ] Update documentation
- [ ] Add/Update data-cy attributes ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))
- [ ] Run `yarn lint`
- [ ] Run `yarn test`
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly
- [ ] Update changelog

breierDemos commented on translations/messages+intl-icu.de.yml:
```diff
@@ -1172,7 +1172,7 @@ error.phone.required: "Bitte geben Sie eine Telefonnummer ein."
 error.enddate.required: "Bitte geben Sie ein Enddatum ein"
 error.date.endbeforestart: "Das Enddatum kann nicht vor dem Startdatum liegen."
-error.phaseIteration.invalid: "Die Durchgangsnummer einer Phase muss eine positive Ganzzahl sein."
+error.phaseIteration.invalid: "Die Durchgangsnummer einer Phase muss eine positive Ganzzahl sein und darf nicht größer als 65.535 sein."
```
The new added validation condition is set to greater than 500. So the error message has to be adjusted too.

breierDemos reviewed (changes requested):
Code Review only

Thank you :)
I assume the 500 is discussed with PO.

There is only a mismatch with your error message.

kijewskidemos commented:
> Code Review only
> 
> Thank you :) I assume the 500 is discussed with PO.
> 
> There is only a mismatch with your error message.

@breierDemos Hey Lukas, thanks for your feedback! I added two translation keys to the code and also adjusted the used method a bit.
